src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs

```diff
@@ -69,6 +71,78 @@ public async Task GetAsync_ConcurrentRead_ThrowsInvalidOperationException()
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [MemberData(nameof(s_versions))]
+        public async Task SendAsync_ServerCertificateValidationCallback_CalledOnce(Version version)
+        {
+            await RemoteExecutor.Invoke(async (version) =>
+            {
+                AppContext.SetSwitch("System.Net.Http.UseWinHttpCertificateCaching", true);
+                int callbackCount = 0;
+                var handler = new WinHttpHandler()
+                {
+                    ServerCertificateValidationCallback = (_, _, _, _) =>
+                    {
+                        Interlocked.Increment(ref callbackCount);
+                        return true;
+                    }
+                };
+                using (var client = new HttpClient(handler))
+                {
+                    for (int i = 0; i < 5; i++)
+                    {
+                        var response = await client.SendAsync(new HttpRequestMessage(HttpMethod.Get, Configuration.Http.SecureRemoteEchoServer)
+                        {
+                            Version = Version.Parse(version)
+                        });
+                        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                        _ = await response.Content.ReadAsStringAsync();
+                    }
+                    Assert.Equal(1, callbackCount);
+                }
+                AppContext.SetSwitch("System.Net.Http.UseWinHttpCertificateCaching", false);
+            }, version.ToString()).DisposeAsync();
+        }
+
+        [OuterLoop]
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [MemberData(nameof(s_versions))]
+        public async Task SendAsync_ServerCertificateValidationCallbackCertificateTimerTriggered_CalledTwice(Version version)
+        {
+            await RemoteExecutor.Invoke(async (version) =>
+            {
+                AppContext.SetSwitch("System.Net.Http.UseWinHttpCertificateCaching", true);
+                int callbackCount = 0;
+                var handler = new WinHttpHandler()
+                {
+                    ServerCertificateValidationCallback = (_, _, _, _) =>
+                    {
+                        Interlocked.Increment(ref callbackCount);
+                        return true;
+                    }
+                };
+                using (var client = new HttpClient(handler))
+                {
+                    var response = await client.SendAsync(new HttpRequestMessage(HttpMethod.Get, Configuration.Http.SecureRemoteEchoServer)
+                    {
+                        Version = Version.Parse(version)
+                    });
+                    Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                    _ = await response.Content.ReadAsStringAsync();
+                    await Task.Delay(65 * 1000); // 65 seconds are enough for the certificate cache to expire.
```

Comment:
Can we avoid doing this?
If the whole thing is behind an AppContext switch, we can consider making the timeout configurable via one as well?

Reply:
Well, either we're gonna delete the test or make it configurable as well, otherwise we can't avoid it.